src/families/tron/types.js

```diff
@@ -13,22 +14,163 @@ export type CoreOperationSpecifics = {};
+  NetUsed: number,
+  NetLimit: number,
+  EnergyLimit: ?number
 |};
 
 export type NetworkInfoRaw = {|
-  family: "tron"
+  family: "tron",
+  freeNetUsed: number,
+  freeNetLimit: number,
+  NetUsed: number,
+  NetLimit: number,
+  EnergyLimit: ?number
 |};
 
 export type Transaction = {|
   ...TransactionCommon,
-  family: "tron"
+  family: "tron",
+  mode: TronOperationMode,
+  resource: ?TronResource,
+  networkInfo: ?NetworkInfo,
+  duration: ?number,
+  votes: Vote[]
 |};
 
 export type TransactionRaw = {|
   ...TransactionCommonRaw,
-  family: "tron"
+  mode: TronOperationMode,
+  family: "tron",
+  resource: ?TronResource,
+  networkInfo: ?NetworkInfoRaw,
+  duration: ?number,
+  votes: Vote[]
+|};
+
+export type TrongridTxType =
+  | "TransferContract"
+  | "TransferAssetContract"
+  | "FreezeBalanceContract"
+  | "UnfreezeBalanceContract"
+  | "VoteWitnessContract"
+  | "TriggerSmartContract"
+  | "WithdrawBalanceContract"
+  | "ExchangeTransactionContract";
+
+export type TrongridTxInfo = {|
+  txID: string,
+  date: Date,
+  type: TrongridTxType,
+  tokenId?: string,
+  from: string,
+  to?: string,
+  value?: BigNumber,
+  fee?: BigNumber,
+  resource?: TronResource
+|};
+
+export type SendTransactionData = {|
+  to_address: string,
+  owner_address: string,
+  amount: number,
+  asset_name: ?string
+|};
+
+export type SmartContractFunction = "transfer(address,uint256)";
+
+export type SmartContractTransactionData = {|
+  function_selector: SmartContractFunction,
+  fee_limit: number,
+  call_value: number,
+  contract_address: string,
+  parameter: string,
+  owner_address: string
+|};
+
+export type UnfreezeTransactionData = {|
+  receiver_address?: string,
+  owner_address: string,
+  resource: ?TronResource
+|};
+
+export type FreezeTransactionData = {|
+  receiver_address?: string,
+  owner_address: string,
+  frozen_balance: number,
+  frozen_duration: number,
+  resource: ?TronResource
+|};
+
+export type SendTransactionDataSuccess = {|
+  raw_data_hex?: string,
+  raw_data: Object,
+  txID: string,
+  signature: ?(string[])
 |};
 
 export const reflect = (_declare: *) => {};
+
+export type SuperRepresentativeData = {|
+  list: SuperRepresentative[],
+  totalVotes: number,
+  nextVotingDate: Date
+|};
+
+export type SuperRepresentative = {|
+  address: string,
+  name: ?string,
+  url: ?string,
+  isJobs: boolean,
+  brokerage: number,
+  voteCount: number,
+  totalProduced: ?number,
+  totalMissed: ?number,
+  latestBlockNum: ?number,
+  latestSlotNum: ?number
+|};
+
+export type TronResources = {|
+  frozen: {
+    bandwidth: ?FrozenInfo,
+    energy: ?FrozenInfo
+  },
+  delegatedFrozen: {
+    bandwidth: ?FrozenInfo,
+    energy: ?FrozenInfo
+  },
+  votes: Vote[],
+  tronPower: number,
+  energy: number,
+  bandwidth: BandwidthInfo,
+  unwithdrawnReward: number
+|};
+
+export type Vote = {|
+  address: string,
+  voteCount: number
+|};
+
+export type FrozenInfo = {|
+  amount: BigNumber,
+  expiredAt: Date
```

Comment:
to me, this is still going to create a problem.
I guess we never know until we try, but I see that `TronResources` is both on Raw and non Raw models. which can't happen. BigNumber and Date can never exists in *Raw models, because they never going to be correctly typed like this when we `JSON.parse` on LLD / LLM side the accounts data